unit_tests/src/test_genquery1_flex_bison_parser.cpp

```diff
@@ -0,0 +1,432 @@
+#include <catch2/catch.hpp>
+
+#include "irods/irods_at_scope_exit.hpp"
+#include "irods/irods_default_paths.hpp"
+#include "irods/rcMisc.h"
+#include "irods/rodsGenQuery.h"
+#include "irods/rodsErrorTable.h"
+
+#include <boost/algorithm/string.hpp>
+#include <fmt/format.h>
+
+#include <cstdlib>
+#include <cstring>
+#include <fstream>
+#include <string>
+#include <utility>
+#include <vector>
+
+// The original GenQuery1 string-based parser.
+// This function is maintained as part of this unit test for backward compatibility reasons.
+auto fillGenQueryInpFromStrCond(char* str, genQueryInp_t* genQueryInp) -> int;
+
+// The following functions are only used by the original GenQuery1 parser.
+// Therefore, they have been pulled out of the public iRODS interface to keep users from using them.
+auto getCondFromString(char* t) -> char*;
+auto goodStrExpr(char* expr) -> int;
+auto separateSelFuncFromAttr(char* t, char** aggOp, char** colNm) -> int;
+
+// NOLINTNEXTLINE(readability-function-cognitive-complexity)
+TEST_CASE("GenQuery1 flex bison parser maintains backward compatibility with original GenQuery1 parser")
+{
+    // IMPORTANT: This test assumes all queries CAN be parsed into a valid GenQuery1
+    // input struct.
+
+    // Open file containing list of GenQuery1 query strings. This list of queries was
+    // produced from the test suite.
+    const auto file_path = irods::get_irods_home_directory() / "unit_tests/genquery1_backwards_compatibility_queries.txt";
```

Comment:
I was wondering whether we need to leave a README or something to indicate how the file was generated, whether it will ever need to be updated (I'm assuming not), etc. Maybe it is sufficient to comment here.

Reply:
I don't think we'll ever need to update or regenerate this file.

I'll add more words about how the file was generated.

Reply:
Done